src/buildfile

```diff
@@ -5,9 +5,9 @@ impl_libs = # Implementation dependencies.
 ./: lib{benchmark} liba{benchmark-main}
 
 pub = [dir_path] ../include
-pub_hdrs = $pub/{$pub_hdrs}
```

Comment:
The main issue is here. I suggest the following code
```
include $pub
pub_hdrs = $pub/{$pub_hdrs}
```
`include` was missing here which actually causes the out of order problem

Reply:
Quick question: 
What exactly does `pub_hdrs = $pub/{$pub_hdrs}` do? Printing this value (or just `$pub_hdrs` prior to assignment) with or without `include($pub)` shows nothing (latter just `[null]`). In fact, if I just include and remove the prerequisite `$pub/$pub_hdrs` from `lib{benchmark}` it makes no difference (it still builds fine).

Reply:
The way I understand it is, `$pub/{$pub_hdrs}` is a scoped access to the variable `$pub_hdrs` defined in `$pub`’s `buildfile`.
Locally assigning it to a variable is only for convenience.
The general idiom I have seen is for `$pub_hdrs` to contain all public headers, usually in `include/`’s `buildfile`. `$pub` or in this case `include/` is responsible for dealing with the headers appropriately for `update`, `dist` and `install`.

> remove the prerequisite $pub/$pub_hdrs from lib{benchmark} it makes no difference (it still builds fine).

The pre-requisite on `lib{benchmark}` is quite important not just for `update`, but other operations like `clean` and `dist` as well. Have a look at [build2 docs, Section 1.1](https://build2.org/build2/doc/build2-build-system-manual.xhtml#intro-hello), which introduces this idea toward the end of the section.

Because this is a package shell of an upstream project that is itself not developed with `build2`, the explicit declaration may not be required here. But in general declaring all pre-requisites, allows `build2` to recognize what files must be built, re-built or cleaned and distributed. 

Not declaring such pre-requisites, caught me off guard in the past, specifically, during `dist`, where source/header seem to be “missing” when building as part of another project’s required package, but everything worked fine when building from a cloned git repo.

In general, best practice is to always declare all target pre-requisites. It makes things clear to both the tool and it’s users.

Reply:
Thank you for the explanation! Also, after messing around a bit it might be the case that `master` is all fine, because what did _catch me off guard_ is the fact that `bdep test` does not imply `bdep update`. So the issues I had that spawned this PR was caused by the fact (I think) that I just ran `bdep test` (which didn't build the `google-benchmark` prerequisite correctly (?). See [this](https://github.com/build2/build2/issues/94) issue for and explanation.

One difference for me though is that `google-benchmark` is a direct prerequisite of my [benchmark test(s)](https://github.com/helmesjo/convertible/blob/02e8c2a2741d24926c1cdf868a5c56a0cf95b606/convertible/buildfile#L62), so I still feel like just running `bdep test` (with `config.convertible.benchmarks=true` to enable those targets) should do the correct thing.

Anyways, it might still be a good thing to explicitly do the `include`, so I'll add only that change.